docs/content/oauth-oidc.md

```diff
@@ -0,0 +1,132 @@
+---
+title: OAuth2 and OpenID Connect
+kind: misc
+weight: 5
+---
+
+In order to make policy decisions we commonly need to know the identity of the caller. Traditionally this was often done by providing a user or client identifier along with the request, and use that identifier to look up further information like user details or permissions from a remote data source. While this model works fairly well for many applications, it scales poorly in distributed systems such as microservice environments. If multiple services are involved serving a single request from a user, having them all ask an external service or database for information about the user would effectively mean doubling the number of total requests sent, and all in order to retrieve the same data for each service!
+
+## JSON Web Tokens
+
+To deal with this, OAuth2 established a basic protocol for obtaining _access tokens_, which would be issued by the authorization server and passed around between services in order to identify and authorize the caller. Using opaque tokens however poses many of the same problems as those faced when passing around user identifiers - rather than asking a database for more information the authorization server would instead have to be consulted whenever additional data was needed. While the OAuth2 specification leaves it to the implementation to decide on the format of the tokens, JSON Web Tokens (JWTs) have emerged as the most common one. Using JWTs as tokens include a number of benefits.
+
+* Self-contained - JWTs contain most, or all, of the details needed to identify the caller.
+* Signed - JWTs are signed by the issuer, meaning their authenticity can be verified by the receiver.
+* Immutable - a JWT can't be changed anywhere in the request chain without invalidating the signature.
+
+The OpenID Connect standard took it one step further, and made JWTs _the_ format for tokens, encoding user identity in a new token type called _ID tokens_. This, along with many other enhancements added a a much welcome identity layer on top of the quite minimal OAuth2 standard. For the purpose of this document, we will deal with both access tokens and ID tokens as JWTs.
```

Comment:
💭 Aren't ID tokens already in the OAuth2 spec, too?

I've always thought the lineage was: OAuth2 defines an _authorization standard_ very broadly; it was commonly (ab)used for _authentication_ means; and OIDC concretised OAuth2 for the authentication use case.

Just a comment, though, nothing that urgently needs changing.

Reply:
Though supposedly an "authorization framework", very little in the spec actually revolves around authorization, at least not in the sense we usually talk about that. _Delegation framework_ would come a lot closer IMO, but oh well.

ID tokens were never in the OAuth2 spec but were introduced with OIDC. Looking at some of the initial work on OAuth3 (which ["does not exist"](https://oauth.net/3/) 😆) it looks like identity might be more integrated in the next iteration of the protocol.

Reply:
Thanks for clarifying this for me! 😃